src/lib.rs

```diff
@@ -312,6 +312,25 @@ impl ArchiveEntryReader {
         Reader { handler: self.handler.clone() }
     }
 
+    pub fn extract_to(self, path : &str) -> Result<Self, ArchiveError> {
+        let extract_path = CString::new(path).unwrap();
+        unsafe {
+            archive_entry_set_pathname(self.entry, extract_path.as_ptr());
+            self.extract()
+        }
+    }
+    pub fn extract(self) -> Result<Self, ArchiveError> {        
+        unsafe {
+          let res = archive_read_extract(*self.handler, self.entry, 0);
```

Comment:
This is incomplete, as I didn't understand the "flags" libarchive expects, and just used a 0 for now.